File: backend/src/main/java/iuh/labbooking/service/otp/OtpService.java
package iuh.labbooking.service.otp;

import iuh.labbooking.dto.request.otp.SendOtpRequest;
import iuh.labbooking.dto.request.otp.VerifyOtpRequest;
import iuh.labbooking.dto.response.otp.SendOtpResponse;
import iuh.labbooking.dto.response.otp.VerifyOtpResponse;

public interface OtpService {

    SendOtpResponse sendOtp(SendOtpRequest request);

    VerifyOtpResponse verifyOtp(VerifyOtpRequest request);

    long getRemainingCooldown(String email, String scope);

    int getRemainingAttempts(SendOtpRequest request);

    void clearOtpData(SendOtpRequest request);

    boolean verifyResetToken(String email, String token);

    void clearAllData(String email);
}

File: frontend/src/services/otp.service.ts
import http from '@/lib/http'
import type { SendOtpRequest, SendOtpResponse, VerifyOtpRequest, VerifyOtpResponse } from '@/schemas/otp.schema'
import type { ApiResponse } from '@/schemas/base.schema'

const otpService = {
  send: (body: SendOtpRequest) => http.post<ApiResponse<SendOtpResponse>>('/otp/send', body),

  verify: (body: VerifyOtpRequest) => http.post<ApiResponse<VerifyOtpResponse>>('/otp/verify', body),

  getCooldown: (email: string, scope: string) =>
    http.get<ApiResponse<number>>(`/otp/cooldown?email=${email}&scope=${scope}`)
}

export default otpService
